### PR TITLE
Remove macros from trans.h/c

### PR DESCRIPTION
**Pull request by james-d-mitchell**
This PR removes almost all of the macros defined and used in the files trans.h/c and pperm.h/c, and replaces them with functions. It also makes it easier to create valid partial perms by not requiring that the codegree of the partial perm is set correctly when the object is created, and the functions replacing macros for partial perms are put into the header so that they can be used elsewhere.

### CODE REVIEW

File: src/trans.c [hunk 1]
```diff
@@ -102,6 +102,34 @@ Obj FuncIMAGE_SET_TRANS(Obj self, Obj f);
...
+{
+    GAP_ASSERT(IS_TRANS(f));
+    GAP_ASSERT(deg == NULL ||
+               (IS_INTOBJ(deg) && INT_INTOBJ(deg) <= DEG_TRANS(f)));
+    if (deg == NULL) {
+        ADDR_OBJ(f)[2] = NULL;
+    }
+    else {
+        ADDR_OBJ(f)[2] = deg;
+    }
```
Comment (fingolfin) [inline src/trans.c]: Why not just `ADDR_OBJ(f)[2] = deg;` instead of that `if/else` ?

File: src/trans.h [hunk 2]
```diff
@@ -1,26 +1,94 @@
...
+
+static inline UInt RANK_TRANS2(Obj f)
+{
+    GAP_ASSERT(TNUM_OBJ(f) == T_TRANS2);
+    return (IMG_TRANS(f) == NULL ? INIT_TRANS2(f) : LEN_PLIST(IMG_TRANS(f)));
+}
+
+static inline Obj NEW_TRANS4(UInt deg)
+{
+    // No assert here since we allow creating new T_TRANS4's when the degree
```
Comment (fingolfin) [inline src/trans.h]: Super nitpick, but: shouldn't it be `T_TRANS4s` (plural) instead of `T_TRANS4's` (possessive) ?

Comment (james-d-mitchell) [inline src/trans.h]: Yup

File: src/trans.c (hunk 1 above)
Comment (james-d-mitchell) [inline src/trans.c]: Agreed

**Comment by james-d-mitchell** [comment]
I think I'll also move some of the new functions into trans.c, since they are not used, or usable elsewhere.

File: src/trans.c (hunk 1 above)
Comment (fingolfin) [inline src/trans.c]: So I'll wait for an update of this PR, then

**Pull request merged**